arbnode/transaction_streamer.go

```diff
@@ -1519,14 +1519,30 @@ func (s *TransactionStreamer) submitEspressoTransactions(ctx context.Context, ig
 	return s.config().EspressoTxnsPollingInterval
 }
 
+func (s *TransactionStreamer) espressoSwitch(ctx context.Context, ignored struct{}) time.Duration {
```

Comment:
instead of checking this in the loop here, should we add the `EnableEspresso` check in batch poster so that it doesnt send any transaction to transaction streamer ( to be further finalized by hotshot) if the flag is not enabled?

Reply:
I think we should add another check to the batch poster. However, I think this loop in the transaction_streamer introduces correct behavior.

If we switch back to the old start method on the `transaction_streamer` and just add the check to the batch poster, any messages that have gotten to the `transaction_streamer`s database to be sent to espresso will still be processed and sent.  I this check in the batch poster will help avoid doing any additional work like attempting to reach espresso API's over the network!

Reply:
Addressed this issue https://github.com/EspressoSystems/nitro-espresso-integration/pull/252/commits/2e898475b32ef1f7039a93a49c903b61d7622a63

Reply:
Thanks, looks good!